### PR TITLE
Added Reactive Programming with Node.js

### PR DESCRIPTION
**Pull request by errazudin**
Reactive Programming with Node.js will show you the paradigms of RP, why you should use it, and the variations available to you

## Hacktoberfest notes:

- due to volume of submissions, we may not be able to review PRs that do not pass tests and do not have informative titles.
- please read our [contributing guidelines](/CONTRIBUTING.md)
- be sure to check the output of Travis-CI for linter errors
- if this is your first open source contribution, make sure it's not your last!

## What does this PR do?
Add Resource(s)

### Description 
Reactive Programming with Node.js will show you the paradigms of RP, why you should use it, and the variations available to you

### Why is this valuable (or not)?
Any back-end developers who understand Node.js or are advanced enough to pick up the basics. Ideal for developers who have an interest in learning about this different programming paradigm that’s being used more and more every day.

### How do we know it's really free?
Yep its free

### For book lists, is it a book? For course lists, is it a course? etc.
Its an ebook

### Checklist:
- [x] Not a duplicate
- [x] Included author(s) if appropriate
- [x] Lists are in alphabetical order
- [x] Needed indications added (PDF, access notes, under construction)

### CODE REVIEW

File: free-programming-books.md [hunk 1]
```diff
@@ -1771,6 +1771,7 @@ Kerridge (PDF) (email address *requested*, not required)
 * [Node.js Notes](http://books.goalkicker.com/NodeJSBook) (PDF)
 * [Node.js Succinctly, Syncfusion](https://www.syncfusion.com/resources/techportal/ebooks/nodejs) (PDF, Kindle) (email address *requested*, not required)
 * [The Node Beginner Book](http://nodebeginner.org)
+* [Reactive Programming with Node.js] (http://www.allitebooks.com/reactive-programming-with-node-js/)
```
Comment (SethiPandi) [inline free-programming-books.md:1774]: Please write the name of the author of the book after.

Remove the space between the ] and ( so it becomes a valid markdown link.

**Review by SethiPandi — COMMENTED** [review]
You can check out the contribution guidelines (CONTRIBUTING.md) for more information on formatting and other guidelines for this repository. You can click on the failed check below and head to Travis CI to see exactly what was wrong with your PR and how to fix it!

You can also look up markdown if you are unfamiliar with the syntax the documents in the repository and using.

**Comment by SethiPandi** [comment]
Also you can replace the spaces with an x in the PR template to check the check boxes!

For example:
* [ ] With a space inside.
* [x] With x inside.

**Review by errazudin — COMMENTED** [review]
update as per commented

File: free-programming-books.md [hunk 2]
```diff
@@ -1771,6 +1771,7 @@ Kerridge (PDF) (email address *requested*, not required)
 * [Node.js Notes](http://books.goalkicker.com/NodeJSBook) (PDF)
 * [Node.js Succinctly, Syncfusion](https://www.syncfusion.com/resources/techportal/ebooks/nodejs) (PDF, Kindle) (email address *requested*, not required)
 * [The Node Beginner Book](http://nodebeginner.org)
+* [Reactive Programming with Node.js](http://www.allitebooks.com/reactive-programming-with-node-js/) - Fernando Doglio (PDF)
```
Comment (SethiPandi) [inline free-programming-books.md:1774]: Sorry, didn't notice the first time I looked!
Could you also keep the lists in alphabetical order?

R comes before T!

**Review by errazudin — COMMENTED** [review]
Ouch my bad. Rearranged as per commented.

**Comment by eshellman** [comment]
This appears to be a spam SEO site.

**Pull request closed without merging**